### DISCUSSION TITLE
Introducing the ZK Credo and ZK Stack

### DISCUSSION THREAD

**Discussion by bxpana** [zkSync-Community-Hub/zksync-developers#38] [Announcements]
## ZK Credo
Sharing the first draft of The ZK Credo — a declaration of the foundational principles for the emerging ZK-powered Internet of Value.

v1 is now live on GitHub for community discussion and feedback from the Credo's rightful owner: You! <https://github.com/zksync/credo>

Every mission requires a North Star, a compass that ensures the voyager stays on course. The ZK Credo serves as this North Star for zkSync, and our hope is that it will fulfill this crucial purpose for the entire web3 ecosystem. 

The ZK Credo addresses (1) principles and (2) community ethos.

(1) The foundational principles that are critical for decentralized networks:

- Trustlessness
- Security
- Reliability
- Censorship-Resistance
- Privacy
- Accessibility
- Hyperscalability
- Sovereignty

(2) The community ethos: The understanding of a shared moral duty for collective action against oppression. Only when this value is deeply embedded within the community can the network withstand the gradual consolidation of power over time. 

## ZK Stack
Meet ZK Stack 🧰: Your modular framework for building sovereign ZK-powered Hyperchains, based on  Era’s open source code.

Last week, we shared an initial draft of the ZK Credo, a set of guiding principles for ZK tech and culture to reshape the internet. The ZK Stack is our contribution to unlocking the ZK-powered Internet of Value and achieving our long-term mission to advance personal freedom for all.

The ZK Stack will give developers full sovereignty – ranging from the choice of the data availability mode to using your project's own token to decentralize the sequencer. It's open source, EVM compatible, battle-tested, and future-proof.

🧠 Learn more: https://blog.matter-labs.io/introducing-the-zk-stack-c24240c2532a

∎  ZK Stack at a glance  ∎

Free: The ZK Stack is developed under free open-source licenses, ensuring its availability at no cost, and we strongly encourage you to contribute and utilize it as you see fit.

Composable: Hyperchains built with the ZK Stack can seamlessly connect in a trustless network, boasting low latency and shared liquidity.

Modular: Customize and shape every aspect of your Hyperchain, from the choice of sequencer and data availability mode to defining your own tokenomics.

Battle-tested: zkSync Era stands as the most widely adopted ZK rollup on Ethereum. With this track record, the ZK Stack is a safe bet with regard to security and reliability.

Future-proof: ZK rollups are the future of Ethereum scaling. To harness the full potential of ZK's superpowers, it is crucial to build the right architecture from the very beginning.

∎  Who is ZK Stack for?  ∎

The ZK Stack will allow teams to build a custom Hyperchain (configured for their specific use case), with fully trustless, cheap, asynchronous connectivity to the rest of the Hyperchain ecosystem.

Hyperchains are especially useful for teams that require:

• A lightweight sequencer with a low latency of sequencing
• Apps that can interact without the trust assumptions of bridges
• A closed private chain, with connectivity to the ecosystem
• A way to use a native token as your system's base token

∎  Get started today  ∎

In the coming weeks, we will be implementing modifications to our codebase to make it easier to checkout, configure, and deploy an instance of the ZK Stack. Our ultimate goal is a one-click deploy!

zkSync is assisting several pilot projects in adopting the ZK Stack; if you wish to be part of this cohort, reach out.

**Comment by bxpana** [comment]
Learn more about the ZK Stack in the Bankless Episode with co-founder Alex G and Head of Engineering Anthony Rose: https://www.youtube.com/watch?v=5PSHz93lWVM

**Reply by Marblez** [reply]
@bxpana Hi, we're thinking about building on the ZK stack and would love to get in touch!

**Reply by bxpana** [reply]
@Marblez apologies for the delay, just seeing this now. If you haven't been in touch with our team yet you can email us at support@zksync.io and we can make sure to get you in touch with our team for more information.

**Reply by trustless2040** [reply]
Hello, I contacted but there is no response. Can you help me with the ZK Stack setup @bxpana

**Reply by bxpana** [reply]
@trustless2040 from what email or domain did you send from?

**Reply by jmwhangjin** [reply]
Nice

**Comment by edevaldonunez** [comment]
This introduction of the ZK Credo and the ZK Stack represents a significant milestone in the evolution of zkSync and the broader ZK-powered Internet of Value. The principles outlined in the ZK Credo provide a solid foundation for decentralized networks, while the ZK Stack offers a modular framework for building sovereign Hyperchains. Developers now have the tools and ethos needed to advance personal freedom and reshape the internet. Exciting times ahead for the web3 ecosystem!

**Comment by jecgaegacharsmi** [comment]
Exciting updates from zkSync! Introducing the ZK Stack and the ZK Credo - foundational principles and a modular framework for building sovereign ZK-powered Hyperchains. Join us on GitHub for discussions and feedback. Learn more about the ZK Stack and how it can empower your projects. Let's shape the future of the ZK-powered Internet of Value together! #ZKStack #ZKCredo #zkSync

**Comment by hunastly** [comment]
Introducing ZK Stack: Your modular framework for building sovereign ZK-powered Hyperchains. Open source, composable, and battle-tested. Learn more: https://blog.matter-labs.io/introducing-the-zk-stack-c24240c2532a 🔍 #ZKStack #ZKPower #Web3 🔗
